plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2926,9 +2926,12 @@ public ConnectorTableHandle makeCompatiblePartitioning(ConnectorSession session,
     @VisibleForTesting
     static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
     {
-        if (partitions.isEmpty()) {
+        if (partitionColumns.isEmpty() && partitions.isEmpty()) {
             return TupleDomain.none();
         }
+        else if (!partitionColumns.isEmpty() && partitions.isEmpty()) {
```

Comment:
add a test to showcase the problem you are fixing.

Reply:
+ add 3 test cases